dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1479,180 +1463,191 @@ synchronized String evaluateInterval() {
-                                 df.format(processCpuUtil) + ", systemCpuUtil: " + df.format(systemCpuUtil));
+                    Tr.event(tc, "deepQueue: " + deepQueue + ", queueDepth: " + queueDepth + ", deepQueueMultiple: "
+                                 + deepQueueMultiple + ", poolIncrement: " + poolIncrement);
                 }
-                /**
-                 * Sleep the controller thread briefly after increasing the pool size
-                 * then update task count before returning to reduce the likelihood
-                 * of a false negative hang check next cycle due to a few non-hung
-                 * tasks executing on the newly created threads
-                 */
-                try {
-                    Thread.sleep(10);
-                } catch (Exception ex) {
-                    // do nothing
+
+                // Handle pausing the task if the pool has been idle
+                if (manageIdlePool(threadPool, deltaCompleted)) {
+                    return "monitoring paused";
                 }
-                completedWork = threadPool.getCompletedTaskCount();
-                return "action taken to resolve hang";
-            }
 
-            if (checkTargetPoolSize(poolSize)) {
-                return "poolSize != targetPoolSize";
-            }
+                if (resolveHang(deltaCompleted, queueEmpty, poolSize, cpuHigh)) {
+                    if (tc.isEventEnabled()) {
+                        Tr.event(tc, "Executor hang detected - poolSize: " + poolSize + ", activeThreads: " + activeThreads +
+                                     ", queueDepth: " + queueDepth + ", cpuUtil: " + df.format(cpuUtil) + ", processCpuUtil: " +
+                                     df.format(processCpuUtil) + ", systemCpuUtil: " + df.format(systemCpuUtil));
+                    }
+                    /**
+                     * Sleep the controller thread briefly after increasing the pool size
+                     * then update task count before returning to reduce the likelihood
+                     * of a false negative hang check next cycle due to a few non-hung
+                     * tasks executing on the newly created threads
+                     */
+                    try {
+                        Thread.sleep(10);
+                    } catch (Exception ex) {
+                        // do nothing
+                    }
+                    completedWork = threadPool.getCompletedTaskCount();
+                    return "action taken to resolve hang";
+                }
 
-            controllerCycle++;
-            ThroughputDistribution currentStats = getThroughputDistribution(poolSize, true);
+                if (checkTargetPoolSize(poolSize)) {
+                    return "poolSize != targetPoolSize";
+                }
 
-            // handleOutliers will mark this 'true' if it resets the distribution
-            distributionReset = false;
+                controllerCycle++;
```

Comment:
```suggestion
                controllerCycle++;
                ThroughputDistribution currentStats = getThroughputDistribution(poolSize, true);
```

Puts it back to where it was.  It makes sense to keep the `distributionReset = false;` right before the `handleOutliers` call since they go together.

Reply:
Done.